libs/cardano-ledger-core/src/Cardano/Ledger/UMap.hs

```diff
@@ -151,6 +155,29 @@ instance DecCBOR RDPair where
   decCBOR =
     decodeRecordNamed "RDPair" (const 2) $ RDPair <$> decCBOR <*> decCBOR
 
+data VDeleg c
+  = CredVoter (Credential 'Staking c)
+  | AbstainRep
+  | NoConfidenceRep
+  deriving (Show, Eq, Ord, Generic, NoThunks, NFData, ToExpr, ToJSON)
```

Comment:
In order to reduce memory overhead and reduce serialization size we should unwind the Credential. Also, lets fixup the names for consistency
```suggestion
data DRep c
  = DRepKeyHash !(KeyHash Voting c)
  | DRepScriptHash !(ScriptHash c)
  | DRepAlwaysAbstain
  | DRepAlwaysNoConfidence
  deriving (Show, Eq, Ord, Generic, NoThunks, NFData, ToExpr, ToJSON)

getDRepCredential :: Drep c -> Maybe (Credential 'Voting c)
getDRepCredential = \case ..
```



Note also it is not `Staking`, but `Voting` :slightly_smiling_face:

Reply:
That will be an extra memory overhead. The idea is that we'll have a lot of them. And since not all of the DRepIds will be in the VState there is no guarantee of memory sharing.

Reply:
In other words unrolling like that will save us a machine word (8 bytes) per DRep that never registered. It will also help with serialization.

Reply:
Do you think it's worth it to hide the unwound credential behind a pattern synonym?

Reply:
Yeah, it might not be a bad idea.